kitsune/sumo/middleware.py

```diff
@@ -103,29 +92,21 @@ def process_request(self, request):
         expiration = request.session.get("oidc_id_token_expiration", 0)
         now = time.time()
         access_token = request.session.get("oidc_access_token")
+        profile = request.user.profile
 
         if access_token and expiration < now:
 
-            response_token_info = (
-                requests.post(settings.FXA_VERIFY_URL, data={"token": access_token})
-            ).json()
-
-            # if the token is not verified, log the user out
-            if (
-                response_token_info.get("code") == 400
-                and response_token_info.get("message") == "Invalid token"
-            ):
-                profile = request.user.profile
-                profile.fxa_refresh_token = ""
-                profile.save()
-                logout(request)
-            else:
-                # we don't need to refresh the token,
-                # an expired token is valid if the FxA session is not
-                # revoked from the FxA settings
+            token_info = FXAAuthBackend.refresh_access_token(profile.fxa_refresh_token)
+            new_access_token = token_info.get("access_token")
+            if new_access_token:
+                request.session["oidc_access_token"] = new_access_token
```

Comment:
For sure, storing a 5-minute access token in the session - which then expires 11 and a bit hours before we hit the refresh route again - is a bit weird, I can see us forgetting this and being confused why calling some other fxa route with `session["oidc_access_token"]` fails in the future.